### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by zhfnjust**

### CODE REVIEW

File: docs/chain_api_zh_CN.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+        script: newLockingScript,
+        satoshis: newAmount,
+        })
+})
+.setInputScript(0, (tx, output) => {  //output 包含输入对应的锁定脚本和 satoshis 余额
+    const preimage = getPreimage(tx, output.script, output.satoshis)
+    const newAmount =  amount - tx.getEstimateFee(); //计算合约的新余额，需要减去交易的费用
+    return counter.unlock(new SigHashPreimage(toHex(preimage)), newAmount).toScript()
+})
+.seal()
```
Comment (xhliu) [inline docs/chain_api_zh_CN.md:105]: //封印交易, 同时自动计算出正确的交易费用和找零余额

File: docs/chain_api_zh_CN.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+### 调用有状态合约
+
+构造调用有状态合约的交易比较复杂。因为解锁脚本的大小影响到交易费用的计算，从而影响输出中的 `satoshis` 余额，输出中的 `satoshis` 余额又会影响 `preimage` 的计算。使用 **scryptlib** 的链式 API 隐藏了处理这些繁琐计算的细节， 你只需按照以下方式来构造交易：
+
+#### 交易费用是由合约支付：
+
+这种情况下合约的新余额在构造交易之前是未知的，交易费用会影响合约的新余额。
+
+1. 通过 `createInputFromPrevTx(tx, outputIndex)` 从合约所在的交易创建输入，并添加到交易中
+2. 使用 `.setOutput(outputIndex, (tx) => bsv.Transaction.Output)` 添加一个有状态合约的输出。合约新的余额通常需要减去交易费用；
```
Comment (xhliu) [inline docs/chain_api_zh_CN.md:69]: 去掉`有状态合约的`

File: docs/chain_api_zh_CN.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+```javascript
+
+const unlockingTx = new bsv.Transaction();
+
+const newAmount = amount + spendAmount； //合约新余额与交易费用无关
+
+unlockingTx.addInput(createInputFromPrevTx(prevTx))
+.from(await fetchUtxos(privateKey.toAddress())) //添加用于支付交易费用的输入
+.addOutput(new bsv.Transaction.Output({ // 添加一个有状态合约的输出, newAmount 是合约新的余额
```
Comment (xhliu) [inline docs/chain_api_zh_CN.md:133]: 去掉`有状态合约的`

File: README.md [hunk 4]
```diff
@@ -267,6 +267,9 @@ let counterClone = Counter.fromHex(counter.lockingScript.toHex());
 
 ```
 
+## Chain Api
```
Comment (xhliu) [inline README.md:270]: Change `C/chain` to `C/chained` to avoid confusion. Chain usually means blockchain in our context.

File: README.md [hunk 5]
```diff
@@ -267,6 +267,9 @@ let counterClone = Counter.fromHex(counter.lockingScript.toHex());
...
+
+[Chain Api](./docs/chain_api_en.md) makes building transactions more efficient and simple
```
Comment (xhliu) [inline README.md:272]: makes building transactions more efficient and simple
->
makes building transactions easy.

File: docs/chain_api_en.md [hunk 6]
```diff
@@ -0,0 +1,162 @@
+# 使用链式 API 构造交易
+
```
Comment (xhliu) [inline docs/chain_api_en.md:2]: Have a English draft after u fix cn version. And I will review again.

**Review by xhliu — CHANGES_REQUESTED** [review]
👍 Looks good overall.

Some minor changes.

File: patches/bsv/lib/errors/spec.js [hunk 7]
```diff
@@ -0,0 +1,187 @@
+'use strict'
+
```
[the remark this answers is not in the document]
Comment (zhfnjust) [reply patches/bsv/lib/errors/spec.js:2]: for add some new Error, like this:

```
{
    name: 'TransactionAlreadySealed',
    message: 'Cannot update sealed transaction'
  }
```

**Review by xhliu — APPROVED** [review]
💯 nice

**Pull request merged**